### PR TITLE
Follow-up to #7252

### PR DESCRIPTION
**Pull request by bgamari**
- Add a changelog entry
 - Carry out a small renaming suggested by @phadej to make the implementation a bit clearer

---
Please include the following checklist in your PR:

*  [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [x] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [ ] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: Cabal/src/Distribution/Simple/GHC.hs [hunk 1]
```diff
@@ -551,13 +551,13 @@ buildOrReplLib mReplFlags verbosity numJobs pkg_descr lbi lib clbi = do
...
   -- modules?
-  let cLikeFiles  = fromNubListR $ mconcat
+  let cLikeSources  = fromNubListR $ mconcat
                       [ toNubListR (cSources   libBi)
                       , toNubListR (cxxSources libBi)
                       , toNubListR (cmmSources libBi)
                       , toNubListR (asmSources libBi)
                       ]
-      cObjs       = map (`replaceExtension` objExtension) cLikeFiles
+      cObjs       = map (`replaceExtension` objExtension) cLikeSources
```
Comment (phadej) [inline Cabal/src/Distribution/Simple/GHC.hs:560]: this is still `cObjs`, which is confusing as it's `cLikeObjs`. Ditto `cProfObjc` etc (which include objects **not only** from C-code, but from **C-like** code, where likeness is vague notion as well).

TL;DR this is not what I suggested.

Comment (emilypi) [reply Cabal/src/Distribution/Simple/GHC.hs:560]: I tend to agree here. Because we don't document, we have to improve the state of variable names. Though, preferably, I'd like both a docstring and better names.

**Comment by emilypi** [comment]
@bgamari CI is fixed if you rebase to master

**Review by emilypi — APPROVED** [review]
